LongestPole/FormatTable.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2024 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey
+-/
+import Mathlib.Data.String.Defs
+
+
+/-!
+# Format Table
+
+This file provides a simple function for formatting a 2d array of strings into a markdown-compliant table.
+-/
+
+inductive Alignment where
+  | left
+  | right
+  | center
+deriving Inhabited, BEq
+
+/--
+Takes a 2d array of string data and renders it into a table.
+ -/
+def formatTable (headers : Array String) (table : Array (Array String))
+    (alignments : Option (Array Alignment) := none) :
+    String := Id.run do
+  -- If no alignments are provided, default to left alignment for all columns
+  let alignments := alignments.getD (Array.mkArray headers.size Alignment.left)
+  -- Get the maximum widths of each column
+  let mut widths := headers.map (·.length)
```

Comment:
Can you use [Array.maxD](https://leanprover-community.github.io/mathlib4_docs/Batteries/Data/Array/Basic.html#Array.maxD) here (with default 0, I guess)?

Reply:
Not sure that that's a good idea. `maxD` operates on arrays, but we want to operate on the columns of the 2d array, so we would have to transpose. Writing it in an imperative style ultimately feels like it will be more understandable.

Reply:
Makes sense - let's leave this as-is.